#! /bin/sh /usr/share/dpatch/dpatch-run
## 26trailing_tab_segv.dpatch by <hesso@pool.math.tu-berlin.de>
##
## DP: Prevent a segfault if a trailing tab is pushed to a new line by inserting
## DP: characters before it if 'set number' is in effect.

@DPATCH@
--- nvi-1.81.6+debian.orig/vi/vs_relative.c	2007-11-18 17:41:42.000000000 +0100
+++ nvi-1.81.6+debian/vi/vs_relative.c	2008-06-13 03:22:55.000000000 +0200
@@ -133,7 +133,7 @@
 	 * Initialize the pointer into the buffer and current offset.
 	 */
 	p = lp;
-	curoff = 0;
+	curoff = scno;
 
 	/* Macro to return the display length of any signal character. */
 #define	CHLEN(val) (ch = *(UCHAR_T *)p++) == '\t' &&			\
